modules/resolver-rule-associations/main.tf

```
@@ -0,0 +1,7 @@
+resource "aws_route53_resolver_rule_association" "this" {
+  for_each = var.create ? var.resolver_rule_associations : tomap({})
```

```suggestion
  for_each = { for k, v in var.resolver_rule_associations : k => v if var.create }
```

This is the right way to write it in `for_each` instead of using `tomap()`. This way it is not going to work always like you expected (in Terraform 0.13 and 0.14+).